evm/src/main/java/org/hyperledger/besu/evm/precompile/P256VerifyPrecompiledContract.java

```diff
@@ -222,7 +222,14 @@ private PrecompileInputResultTuple computeDefault(final Bytes input) {
       return new PrecompileInputResultTuple(
           enableResultCaching ? input.copy() : input, PrecompileContractResult.success(INVALID));
     }
-
+    try {
+      final org.bouncycastle.math.ec.ECPoint ecPoint = R1_PARAMS.getCurve().createPoint(qx, qy);
+      signatureAlgorithm.getCurve().validatePublicPoint(ecPoint);
```

Comment:
There is not much documentation on the method, but looking at the code it should be the correct check:

https://github.com/bcgit/bc-java/blob/18692ff1e92f6fe81bf92f4b48da3a6f32c1d0d4/core/src/main/java/org/bouncycastle/crypto/params/ECDomainParameters.java#L157